test/table_test.rb

```diff
@@ -15,4 +15,30 @@
       table.candidate_parent_classes.must_equal []
     end
   end
+
+  describe "#" do
+    let(:table) { PartialKs::Table.new(PostTag) }
+
+    it "emits the tag_id condition for belongs_to" do
+      table = PartialKs::Table.new(BlogPost)
+      table.relation_for_association(User).to_sql.must_equal "SELECT \"blog_posts\".* FROM \"blog_posts\" WHERE \"blog_posts\".\"user_id\" = 0"
+    end
+
+    it "emits the subquery for has_many" do
+      table = PartialKs::Table.new(BlogPost)
+      post_tag_id = 111
+
+      PostTag.stub :pluck, [post_tag_id] do    # TODO verify the pluck
+        table.relation_for_association(PostTag).to_sql.must_equal("SELECT \"blog_posts\".* FROM \"blog_posts\" WHERE \"blog_posts\".\"id\" IN (0, #{post_tag_id})")
+      end
+    end
+
+    describe "table with different :foreign_key" do
+      let(:table) { PartialKs::Table.new(OldTag) }
+
+      it "uses the foreign key that's present in the table" do
+        table.relation_for_association(OldEntry).to_sql.must_equal "SELECT \"old_tags\".* FROM \"old_tags\" WHERE \"old_tags\".\"blog_post_id\" = 0"
```

Comment:
These SELECT queries feel quite brittle. No good ideas as to alternatives currently but it does lock one into the vagaries of how ActiveRecord emits SQL statements. Is a small hassle to update tests, is all